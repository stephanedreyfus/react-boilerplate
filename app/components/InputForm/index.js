/**
 * InputForm
 * Takes in phrase and passes it up to container/parent to send to store.
 */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
  Form,
  Label,
  Input,
  PhraseButton,
  LabelInputWrapper,
} from '../Styling/InputStyle';

// import { FormattedMessage } from 'react-intl';
// import messages from './messages';

class InputForm extends PureComponent {
  static propTypes = { addPhrase: PropTypes.func };

  constructor(props) {
    super(props);

    this.state = {
      phrase: '',
    };

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(e) {
    e.preventDefault();
    // Does not accept phrase if it contains only spaces.
    if (this.state === undefined || this.state.phrase.trim === undefined) {
      // FIXME Make this a custom alert or a modal
      alert('Please use at least one character or number.');
    } else {
      this.props.addPhrase(this.state.phrase);
    }
  }

  handleChange(e) {
    this.setState({ phrase: e.target.value });
  }

  render() {
    return (
      <Form>
        <LabelInputWrapper>
          <Label htmlFor="phrase">Enter a Phrase Below</Label>
          <Input
            id="phrase"
            type="text"
            placeholder="Write anything at all!"
            value={this.state.phrase}
            onChange={this.handleChange}
          />
        </LabelInputWrapper>
        <PhraseButton onClick={this.handleSubmit}>Add Phrase!</PhraseButton>
      </Form>
    );
  }
}

export default InputForm;
