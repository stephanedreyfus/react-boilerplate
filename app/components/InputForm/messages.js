/*
 * InputForm Messages
 *
 * This contains all the text for the InputForm component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.InputForm';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the InputForm component!',
  },
});
