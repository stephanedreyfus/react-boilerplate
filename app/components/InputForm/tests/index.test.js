/**
 *
 * Tests for InputForm
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import Enzyme, { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import InputForm from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

Enzyme.configure({ adapter: new Adapter() });

describe('<InputForm />', () => {
  let wrapper;
  let phraseInput;

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <InputForm />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Should change "phrase" in state', () => {
    wrapper = mount(<InputForm />);
    phraseInput = wrapper.find('#phrase').at(2);
    phraseInput.instance().value = 'test phrase';
    phraseInput.simulate('change');

    expect(wrapper.state().phrase).toEqual('test phrase');
  });

  // FIXME Still debugging why the mocked submit will not function.
  it('runs a mocked fn on submit', () => {
    const submitFn = jest.fn();
    wrapper = mount(<InputForm onClick={submitFn} />);
    const form = wrapper.find('form');
    form.simulate('submit');

    expect(submitFn).toHaveBeenCalled();
  });
});
