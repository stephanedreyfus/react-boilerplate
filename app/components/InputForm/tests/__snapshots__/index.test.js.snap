// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputForm /> matches snapshot 1`] = `undefined`;
